spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/watcher/DependencyStateChangeListenerRegistry.java

```diff
@@ -0,0 +1,47 @@
+package org.springframework.cloud.zookeeper.discovery.watcher;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.state.ConnectionState;
+import org.apache.curator.x.discovery.ServiceCache;
+import org.apache.curator.x.discovery.details.ServiceCacheListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class DependencyStateChangeListenerRegistry implements ServiceCacheListener {
+
+    private static final Logger log = LoggerFactory.getLogger(DependencyStateChangeListenerRegistry.class);
+
```

Comment:
Ok tried to fix the thread safety like this https://github.com/marcingrzejszczak/spring-cloud-zookeeper/pull/1/files . If you approve then I'll merge it and we will see that in this PR.